File: 0908-middle-of-the-linked-list/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        count = 0
        current = head
        while current:
            count += 1
            current = current.next
        current = head
        for i in range(count // 2):
            current = current.next
        return current

File: 0908-middle-of-the-linked-list/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* middleNode(struct ListNode* head){
    struct ListNode * slow = head, *fast = head->next;
    while (fast) {
        slow = slow->next;
        if (!fast->next)
            break;
        fast = fast->next->next;
    }
    return slow;
}
